NineChronicles.Headless.Executable/Commands/GenesisCommand.cs

```diff
@@ -14,125 +11,210 @@
 
 namespace NineChronicles.Headless.Executable.Commands
 {
     public class GenesisCommand : CoconaLiteConsoleAppBase
     {
-        private static readonly Codec Codec = new Codec();
+        private const int DefaultCurrencyValue = 10000;
+        private static readonly Codec _codec = new Codec();
         private readonly IConsole _console;
 
         public GenesisCommand(IConsole console)
         {
             _console = console;
         }
 
-        [Command(Description = "Mine a new genesis block.")]
-        public void Mine(
-            [Argument("CONFIG", Description = "JSON config path to mine genesis block.")]
-            string configPath
+        private void ProcessData(DataConfig config, out Dictionary<string, string> tableSheets)
+        {
+            Console.WriteLine("Processing data for genesis...");
+            if (string.IsNullOrEmpty(config.TablePath))
+            {
+                throw Utils.Error("TablePath is not set.");
+            }
+
+            tableSheets = Lib9cUtils.ImportSheets(config.TablePath);
+        }
+
+        private void ProcessCurrency(
+            CurrencyConfig? config,
+            out PrivateKey initialMinter,
+            out List<GoldDistribution> initialDepositList
         )
         {
-            var options = new JsonSerializerOptions
+            Console.WriteLine("Processing currency for genesis...");
+            if (config is null)
             {
-                AllowTrailingCommas = true,
-                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
-            };
+                Log.Information("CurrencyConfig not provided. Skip setting...");
+                initialMinter = new PrivateKey();
+                initialDepositList = new List<GoldDistribution>
+                {
+                    new()
+                    {
+                        Address = initialMinter.ToAddress(), AmountPerBlock = DefaultCurrencyValue,
+                        StartBlock = 0, EndBlock = 0
+                    }
+                };
+                return;
+            }
 
-            string json = File.ReadAllText(configPath);
-            GenesisConfig genesisConfig = JsonSerializer.Deserialize<GenesisConfig>(json, options);
-            if (string.IsNullOrEmpty(genesisConfig.TablePath))
+            if (string.IsNullOrEmpty(config.Value.InitialMinter))
             {
-                throw Utils.Error("TablePath is not set.");
+                Log.Information("Private Key not provided. Create random one...");
+                initialMinter = new PrivateKey();
+            }
+            else
+            {
+                initialMinter = new PrivateKey(config.Value.InitialMinter);
             }
 
-            Dictionary<string, string> tableSheets = Lib9cUtils.ImportSheets(genesisConfig.TablePath);
+            initialDepositList = new List<GoldDistribution>();
+            if (config.Value.InitialCurrencyDeposit.Count == 0)
```

Comment:
It seems to assume the `InitialCurrencyDeposit` exists. Then is it also required?

Reply:
I'll make it optional. Thanks for pointing.